test/libsolidity/semanticTests/externalContracts/base64.sol

```diff
@@ -51,10 +51,10 @@ contract test {
 // encode_no_asm(bytes): 0x20, 5, "fooba" -> 0x20, 8, "Zm9vYmE="
 // encode_no_asm(bytes): 0x20, 6, "foobar" -> 0x20, 8, "Zm9vYmFy"
 // encode_inline_asm_large()
-// gas irOptimized: 1385042
+// gas irOptimized: 1368042
 // gas legacy: 1652033
 // gas legacyOptimized: 1201033
 // encode_no_asm_large()
-// gas irOptimized: 3335099
+// gas irOptimized: 3291092
```

Comment:
I see that even with this PR and relaxed inlining the non-asm base64 implementation is still far behind OZ's [assembly implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2884).

Reply:
The main part is checked arithmetic on the index increments... if I put a large ``unchecked`` block in the body of ``NoAsmBase64.encode``, then the gas costs here reduce by 60% down to ``1911077``... which is still worse than the assembly case above, but at least in the same league...
Might be a nice case for adding a user-defined-value type for encapsulating unchecked arithmetic for those indices...

Reply:
Or we could add unchecked operators :)